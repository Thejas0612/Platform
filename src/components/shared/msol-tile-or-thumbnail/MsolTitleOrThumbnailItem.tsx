import React, { FC, MouseEventHandler } from "react";
import { Box, Checkbox, Stack, Typography, useTheme } from "@mui/material";
import { grey } from "@mui/material/colors";
import { Info } from "@mui/icons-material";
import Tooltip from "@mui/material/Tooltip";
import IconButton from "@mui/material/IconButton";

function findBorderColor(checked: boolean, error: boolean): string {
  if (error) {
    return "var(--ddl-color--primary-red)";
  }
  if (checked) {
    return "var(--ddl-color--primary-emerson-green)";
  }

  return grey[200];
}

export interface MsolTitleOrThumbnailItemProps {
  checked: boolean
  onChange?: MouseEventHandler<HTMLDivElement>
  imgUrl: string,
  disabled?: boolean
  title: string
  description: string,
  tooltip?: string,
  hideCheckbox?: boolean
  showError: boolean,
}

export const MsolTitleOrThumbnailItem: FC<MsolTitleOrThumbnailItemProps> = ({
                                                                              checked,
                                                                              onChange = () => {
                                                                              },
                                                                              imgUrl,
                                                                              disabled = false,
                                                                              title,
                                                                              description,
                                                                              showError = false,
                                                                              tooltip,
                                                                              hideCheckbox = false
                                                                            }) => {
  const theme = useTheme();
  const colorOverride = disabled ? theme.palette.text.disabled : undefined;

  const handleChange: MouseEventHandler<HTMLDivElement> = (event) => {
    if (disabled) {
      return;
    }

    onChange(event);
  };

  return <Box
    onClick={handleChange}
    sx={{
      cursor: !disabled ? "pointer" : undefined,
      padding: ".5rem",
      border: 1,
      borderColor: findBorderColor(checked, showError)
    }}
  >
    <Stack direction="row"
           alignItems="center">

      {!hideCheckbox && <Checkbox checked={checked} disabled={disabled} />}

      <Box
        component="img"
        sx={{
          height: { xs: "6rem", md: "9rem;" },
          width: "auto"
        }}
        src={imgUrl}
        alt={title}
      />

      <Box sx={{ margin: 2 }}>
        <>
          <Typography gutterBottom variant="h5" sx={{ color: colorOverride }}>
            {title}
            {
              tooltip != null && <Tooltip title={tooltip} placement="top" arrow>
                <IconButton>
                  <Info sx={{ color: colorOverride || "primary" }} />
                </IconButton>
              </Tooltip>
            }
          </Typography>

          <Typography variant="body2" sx={{ color: colorOverride }}>
            {description}
          </Typography>
        </>
      </Box>
    </Stack>
  </Box>;
};