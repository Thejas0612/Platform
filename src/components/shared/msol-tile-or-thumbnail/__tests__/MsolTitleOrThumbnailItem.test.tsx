import { MsolTitleOrThumbnailItem, MsolTitleOrThumbnailItemProps } from "../MsolTitleOrThumbnailItem";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

const PROPS: MsolTitleOrThumbnailItemProps = {
  checked: false,
  imgUrl: "test url",
  title: "test title",
  description: "test description",
  showError: false
};

describe("<MsolTitleOrThumbnailItem {...PROPS} />", () => {
  test("when disabled anc clicked, then prevent onChange event", async () => {
    const handleChange = jest.fn();
    render(<MsolTitleOrThumbnailItem {...PROPS} disabled={true} onChange={handleChange} />);

    const title = screen.getByText("test title");
    await userEvent.click(title);

    expect(handleChange).not.toHaveBeenCalled();
  });

  test('when tooltip is empty string, then hide tooltip', () => {
    const {container} =  render(<MsolTitleOrThumbnailItem {...PROPS} tooltip={""} />);

    expect(container).toMatchSnapshot()
  })
});