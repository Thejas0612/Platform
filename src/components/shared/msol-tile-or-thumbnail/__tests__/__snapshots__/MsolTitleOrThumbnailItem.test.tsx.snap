// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MsolTitleOrThumbnailItem {...PROPS} /> when tooltip is empty string, then hide tooltip 1`] = `
<div>
  <div
    class="MuiBox-root css-y8llyz"
  >
    <div
      class="MuiStack-root css-1d9cypr-MuiStack-root"
    >
      <span
        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
      >
        <input
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxOutlineBlankIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      <img
        alt="test title"
        class="MuiBox-root css-1s768nl"
        src="test url"
      />
      <div
        class="MuiBox-root css-1c5ij41"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
        >
          test title
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          test description
        </p>
      </div>
    </div>
  </div>
</div>
`;
